cg/meta/encryption/encryption.py

```diff
@@ -92,6 +131,160 @@ def get_symmetric_decryption_command(
+        self.encryption_dir: Path = encryption_dir
+        self.tar_api: TarAPI = tar_api
+        self.pigz_binary_path: str = pigz_binary_path
+        self.slurm_api: SlurmAPI = slurm_api
+        self.slurm_account: str = sbatch_parameter.get("account")
+        self.slurm_hours: int = sbatch_parameter.get("hours")
+        self.slurm_mail_user: str = sbatch_parameter.get("mail_user")
+        self.slurm_memory: int = sbatch_parameter.get("memory")
+        self.slurm_number_tasks: int = sbatch_parameter.get("number_tasks")
+        self.slurm_api.set_dry_run(dry_run=self.dry_run)
+
+    @property
+    def flow_cell_encryption_dir(self) -> Path:
+        return Path(self.encryption_dir, self.flow_cell.full_name)
+
+    @property
+    def flow_cell_encrypt_file_path_prefix(self) -> Path:
+        return Path(self.flow_cell_encryption_dir, self.flow_cell.id)
+
+    @property
+    def pending_file_path(self) -> Path:
+        return Path(self.flow_cell_encrypt_file_path_prefix.with_suffix(FileExtensions.PENDING))
+
+    @property
+    def complete_file_path(self) -> Path:
+        return Path(self.flow_cell_encrypt_file_path_prefix.with_suffix(FileExtensions.COMPLETE))
+
+    def get_flow_cell_symmetric_encryption_command(
+        self, output_file: Path, passphrase_file_path: Path
+    ) -> str:
+        """Generates the Gpg command for symmetric encryption of file."""
+        encryption_parameters: List[str] = (
+            [self.binary_path]
+            + GPGParameters.SYMMETRIC_ENCRYPTION
+            + [passphrase_file_path.as_posix()]
+            + GPGParameters.OUTPUT_PARAMETER
+            + [output_file.as_posix()]
+        )
+        return " ".join(encryption_parameters)
+
+    def get_flow_cell_symmetric_decryption_command(
+        self, input_file: Path, passphrase_file_path: Path
+    ) -> str:
+        """Generates the Gpg command for symmetric decryption."""
+        decryption_parameters: List[str] = (
+            [self.binary_path]
+            + GPGParameters.SYMMETRIC_DECRYPTION
+            + [passphrase_file_path.as_posix(), input_file.as_posix()]
+        )
+        return " ".join(decryption_parameters)
+
+    def is_encryption_possible(self) -> Tuple[bool, Union[str, None]]:
+        """Check if requirements for encryption are meet."""
+        if not self.flow_cell.is_flow_cell_ready():
+            return False, f"Flow cell: {self.flow_cell.id} is not ready"
+        if self.complete_file_path.exists():
+            return False, f"Encryption already completed for flow cell: {self.flow_cell.id}"
+        if self.pending_file_path.exists():
+            return False, f"Encryption already started for flow cell: {self.flow_cell.id}"
+        return True, None
+
+    def encrypt_flow_cell(
+        self,
+    ) -> None:
+        """Encrypt flow cell via GPG and SLURM."""
+        self.flow_cell_encryption_dir.mkdir(exist_ok=True, parents=True)
+        self.create_pending_file(pending_path=self.pending_file_path)
```

Comment:
Should this not be done before the encrypt call? Seems to me that this could be against SRP

Reply:
Fixed